spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala

```diff
@@ -0,0 +1,257 @@
+
+import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.{Column, DataFrame, SparkSession}
+import za.co.absa.atum.AtumImplicits
+import za.co.absa.atum.core.Atum
+import za.co.absa.enceladus.common.RecordIdGeneration.getRecordIdGenerationStrategyFromConfig
+import za.co.absa.enceladus.common.config.{JobConfig, PathConfig}
+import za.co.absa.enceladus.common.plugin.menas.MenasPlugin
+import za.co.absa.enceladus.common.{CommonJobExecution, Constants}
+import za.co.absa.enceladus.dao.MenasDAO
+import za.co.absa.enceladus.dao.auth.MenasCredentials
+import za.co.absa.enceladus.model.Dataset
+import za.co.absa.enceladus.standardization.config.{StandardizationConfig, StandardizationConfigInstance}
+import za.co.absa.enceladus.standardization.interpreter.StandardizationInterpreter
+import za.co.absa.enceladus.standardization.interpreter.stages.PlainSchemaGenerator
+import za.co.absa.enceladus.utils.fs.FileSystemVersionUtils
+import za.co.absa.enceladus.utils.modules.SourceId
+import za.co.absa.enceladus.utils.performance.PerformanceMetricTools
+import za.co.absa.enceladus.utils.schema.{MetadataKeys, SchemaUtils, SparkUtils}
+import za.co.absa.enceladus.utils.udf.UDFLibrary
+import za.co.absa.enceladus.utils.validation.ValidationException
+
+import scala.util.control.NonFatal
+
+trait StandardizationExecution extends CommonJobExecution {
+  private val sourceId = SourceId.Standardization
+
+  protected def prepareStandardization[T](args: Array[String],
+                                          menasCredentials: MenasCredentials,
+                                          preparationResult: PreparationResult
+                                         )
+                                         (implicit dao: MenasDAO,
+                                          cmd: StandardizationConfig[T],
+                                          fsUtils: FileSystemVersionUtils,
+                                          spark: SparkSession): StructType = {
+    // Enable Menas plugin for Control Framework
+    MenasPlugin.enableMenas(
+      conf,
+      cmd.datasetName,
+      cmd.datasetVersion,
+      cmd.reportDate,
+      preparationResult.reportVersion)
+
+    // Enable Control Framework
+    import za.co.absa.atum.AtumImplicits.SparkSessionWrapper
+    spark.enableControlMeasuresTracking(s"${preparationResult.pathCfg.inputPath}/_INFO")
+      .setControlMeasuresWorkflow(sourceId.toString)
+
+    // Add report date and version (aka Enceladus info date and version) to Atum's metadata
+    Atum.setAdditionalInfo(Constants.InfoDateColumn -> cmd.reportDate)
+    Atum.setAdditionalInfo(Constants.InfoVersionColumn -> preparationResult.reportVersion.toString)
+
+    // Add the raw format of the input file(s) to Atum's metadata as well
+    Atum.setAdditionalInfo("raw_format" -> cmd.rawFormat)
+
+    PerformanceMetricTools.addJobInfoToAtumMetadata("std", preparationResult.pathCfg.inputPath, preparationResult.pathCfg.outputPath,
```

Comment:
In standardization, this is in the `prepare` stage, in conformance at the end. Why?

Reply:
It used to be like this before, but I cannot say why. Anyone has any idea?

Reply:
@yruslan Can we pull you into this?

Reply:
Sorry it took so long. I was checking and remembering. :)

There 2 methods that add Atum metadata: `addJobInfoToAtumMetadata()` and `addPerformanceMetricsToAtumMetadata()`. Originally it was 1 method that was invoked at the end. But we encountered several situations when a job has failed and we wanted to investigate. But since Atum's metradata was written at the end of each job, many very helpful fields like 'rawFormat'  were missing. So we split the method into 2, one is invoked at the beginning, and one at the end.

Later, probably due to another refactoring, Standardization anc Conformance were refactored slightly differently, so  `addJobInfoToAtumMetadata()` ended up in 2 different methods.

But basically, it should look like
```scala
PerformanceMetricTools.addJobInfoToAtumMetadata(...)
runTheJob()
PerformanceMetricTools.addPerformanceMetricsToAtumMetadata(...)
``` 

Maybe the code could be refactored so that these 3 steps are more clear. And also, method names could be renamed to reflect the intention. Like `addInitialJobMetadataToAtum()` and `addFInalJobMetadataToAtum()`, or something to that effect.

Reply:
```suggestion
    PerformanceMetricTools.addJobInfoToAtumMetadata("standardization", preparationResult.pathCfg.inputPath, preparationResult.pathCfg.outputPath,
```